src/sentry/auth/providers/fly/provider.py

```diff
@@ -82,3 +82,7 @@ def build_identity(self, state):
             "data": self.get_oauth_data(data),
             "email_verified": False,
         }
+
+
+class FlyOAuthPartnershipProvider(FlyOAuth2Provider):
```

Comment:
nit: `DisableableFlyOAuth2Provider` maybe or `ConfigurableFlyOAuth2Provider`? idk names are hard but ideally something that'd make it clear that this provider can be disabled